### ISSUE TITLE
Doesn't allow installation of local packages

### ISSUE THREAD

**Issue by najamkhn** [yarnpkg/yarn#2767]
It seems like Yarn doesn't allow installation of unpublished packages. Especially if the package is fetched from a private repository?

**Comment by bestander** [comment]
Please provide clarification with an example

**Comment by briandipalma** [comment]
We are seeing this in our company repo

```bash
10:46 $ yarn
yarn install v0.21.3
info No lockfile found.
warning mobile@3.3.0: No license field
[1/4] 🔍  Resolving packages...
Couldn't find any versions for "array-find" that matches "../../packages/array-find"
? Please choose a version from this list: (Use arrow keys)
❯ 0.1.0
  0.0.1
  1.0.0
Couldn't find any versions for "amcharts" that matches "../../packages/amcharts"
? Please choose a version from this list: (Use arrow keys)
Couldn't find any versions for "appcache" that matches "../../packages/appcache"
? Please choose a version from this list: (Use arrow keys)
❯ 0.0.1
  0.0.3
  0.0.2
Couldn't find any versions for "appconsole" that matches "../../packages/appconsole"
```

@najamkhn Could you give an example of your output?

It appears that if you have a local dependency that matches an npm package name yarn gets confused and tries to install the dependency from the npm repo (in our case the company artifactory server)?

**Comment by najamkhn** [comment]
Hey, in my case, here's the output that I get:
```
>>> yarn install
yarn install v0.20.3
info No lockfile found.
[1/4] 🔍  Resolving packages...
error Couldn't find package "exPlayer" on the "npm" registry.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

exPlayer exists in a private repository for us and in the package json we've it something like this:
```
...
"exPlayer":"git@bitbucket.org:ex-player/ex-player.git#master"
...
```

It does resolve it properly if I do `npm install`.

**Comment by rally25rs** [comment]
I know this issue is a bit old now, but try adding a protocol before the URI:

> git+ssh://git@bitbucket.org:ex-player/ex-player.git#master

**Comment by briandipalma** [comment]
For what it's worth our issue has been fixed, we are now using Yarn.

**Comment by rally25rs** [comment]
Closing as this should be fixed in latest Yarn (1.3.2).
